### PR TITLE
Dockerize and add build artifacts

### PR DESCRIPTION
**Pull request by colinxfleming**
Adds our build junk and Dockerizes, so we can stitch together with compose or ansible. This'll take a `.env` file and throw whatever shiny server is running on 3838 behind an auth0 wall.

@cwegrzyn @curtismorales for approval please.

### CODE REVIEW

**Comment by colinxfleming** [comment]
(You can still apparently go straight to `localhost:3838/sampler` if it's accessible, so there's some devops setup we need to do there once this is out in the world, but I assume there's some magic we can do to restrict that a bit once it's on an ec2 box.)

**Comment by cwegrzyn** [comment]
Actually, along those lines, mind rigging up a [docker compose file](https://docs.docker.com/compose/overview/) that runs both this and the sampler? You can basically replace run.sh (plus the additional step of running the sampler) with it.

**Comment by colinxfleming** [comment]
@cwegrzyn docker-compose file rigged up, run file deleted. Tested and it works for me, let me know if you'd like any other changes?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  ports:
+    - "3000:3000"
+  links:
+    - sampler
+  env_file:
+    - .env
+sampler:
+  image: docker.int.bluelabs.io/bluelabsio/sampler:latest
+  ports:
+    - "3838:3838"
```
Comment (cwegrzyn) [inline docker-compose.yml]: You should be all to drop this, and for `web` section set the SHINY_HOST env var to `sampler`

**Comment by colinxfleming** [comment]
Made the commit but it doesn't seem to be linking the sampler hostname properly, and instead thinks it's a string instead of using the sampler container's hostname. Couldn't figure it out in a half hour of googling, I'll take a stab again early tomorrow and follow up with you if it goes nowhere.

**Comment by cwegrzyn** [comment]
Hey— not sure I understand the behavior you are describing exactly, but to
be clear: with that docker compose, the sampler hostname IS “sampler”.
Docker configures name resolution in each container so that the link name
resolves to the host of the linked container.

**Comment by colinxfleming** [comment]
Not sure I have the vocabulary to explain what I'm seeing here -- the js app seems to want everything on localhost and with exposed ports, and gets weird if it's not? I'll hit you up when the dust settles a bit this morning if I can't figure it out by lunch or so.

**Comment by colinxfleming** [comment]
Here we go (thanks @vinceatbluelabs ) -- 

As of 2e24191 here's the behavior of `docker-compose up` --

* Going to `localhost:3000` sends you to an auth0 login page, and then transfers you to `localhost:3000/sampler/`, home of the sampler
* Going to `localhost:3000/sampler/` redirects you to the login page if you aren't logged in
* Going to `localhost:3838/sampler/` errors

I think that's what we're gunning for, no?

**Pull request merged**